src/iam/permhandler/permhandler.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/iam/permhandler.hpp"
+#include "log.hpp"
+
+namespace aos {
+namespace iam {
+namespace permhandler {
+
+/***********************************************************************************************************************
+ * Public
+ **********************************************************************************************************************/
+
+RetWithError<StaticString<uuid::cUUIDStrLen>> PermHandler::RegisterInstance(
+    const InstanceIdent& instanceIdent, const Array<FunctionalServicePermissions>& instancePermissions)
+{
+    LockGuard lock(mMutex);
+
+    Error                           err;
+    StaticString<uuid::cUUIDStrLen> uuidSecret;
+
+    Tie(uuidSecret, err) = GetSecretForInstance(instanceIdent);
+    if (err.IsNone()) {
+        LOG_DBG() << "Instance " << instanceIdent << " has been already registered";
+
+        return {uuidSecret, err};
+    }
+
+    uuidSecret = GenerateSecret();
+    err        = AddSecret(uuidSecret, instanceIdent, instancePermissions);
+
```

Comment:
Remove empty line:
```c++
err = ...
if (!err.IsNone) {
...
}
```
it is common template

Reply:
Done